### DISCUSSION TITLE
Does Oqtane fit for this scenario?

### DISCUSSION THREAD

**Discussion by sturlath** [oqtane/oqtane.framework#3609] [Q&A]
I´m going to create a mulititenant SaaS service that has the following requirements and was wondering if Oqtane is a good fit or not for.

- Tenants can sign up and have a subdomain site that they can sign theyr own DNS url to
   - Tenants should be able to manage many such sites with one login
- Tenants can add and remove menuitems that have pages (Home/About us/blog etc) for their own site. 
- Tenants should be able to change the look and feel of the page (change colors, set darktheme or even change/choose totally different theme or even customize one them selfs)
- Each site should be able to translate all content

**Comment by thabaum** [comment]
If I understand the requirements and if you understand Oqtane Framework is a developers framework that allows you to create all kinds of custom business operations.   All these features you may find already developed by the community just visit the [Oqtane Marketplace](https://www.oqtane.net) to find potential solutions and ideas.  You can possibly put in requests for features or modules here and some community developer may decide to make something that fits your needs.  If you have an issue trying to make something work the community loves to support as it can.

You may have to create your own signup onboarding process.  In the Oqtane Framework as far as I know I am pretty sure you can create domains that use sub domains or whatever url or "Alias" you want and each tenant or "Site" can be administrated by different users assigned to the "Administrator" role.  Using roles and claims you should be able to control who can do what.  With the open source projects you can pretty much patch something off of what already exists.

"Site Groups" has been a topic of the past of a feature to enhance Oqtane at some point.  I would inspect it's behavior and see how you may modify it to fit your needs.

In the [Oqtane Marketplace](https://www.oqtane.net) there is an Oqtane Blog module and others you can search for, test and use as a starting point to enhance as needed.

If you are site administrator you get all those features and then some with the permissions built in the Oqtane Framework.

Themes can be created using the Oqtane Theme as a starting point and enhance it with features as desired or look at the [Oqtane Marketplace](https://www.oqtane.net) for new themes as they are released by the community.

Oqtane has the ability to allow users to change languages and these translations use language resource files which can be edited as desired in all languages including English for most all parts of the framework.  There are a few community supported languages again in the [Oqtane Marketplace](https://www.oqtane.net).

Contributions towards helping everyone reach our goals are very welcome so as you learn more if you find something you would like to contribute, teamwork is what makes the dream work!  I believe you should give it a try to see how far it gets you down the road.  Let us know if you hit a bump.

I hope this helps, cheers!

**Reply by sturlath** [reply]
Thanks for the answer, its giving me a good starting point. But yes I think I just need to get my hand dirty and figure some of these things out for my self. I have been writing down the requiremnts and the list just gets longer and longer while I dig deeper... but yes I´ll for sure let you know when I hit a bump 😅hopefully I can give back at some point.

**Comment by sbwalker** [comment]
@sturlath Oqtane is multi-tenant and supports both shared and isolated tenancy models for data. In regards to your other questions...

- Tenants can sign up and have a subdomain site that they can sign theyr own DNS url to
>> Yes, you can have a single installation of Oqtane that contains multiple sites. Each site has can have its own domain, subdomain, or even subfolder (ie. domain.com/folder). Only the "host" user in an installation is allowed to create new sites.
- Tenants should be able to manage many such sites with one login
>> If the sites for a client are all served from the same database, then the client can have a single login
-  Tenants can add and remove menuitems that have pages (Home/About us/blog etc) for their own site.
>> An Administrator of a site can add pages and content
- Tenants should be able to change the look and feel of the page (change colors, set darktheme or even change/choose totally different theme or even customize one them selfs)
>> themes define the look and feel. The host user defines the themes which are available to sites.
- Each site should be able to translate all content
>> you would need to create a separate site for each language ie. tenant1.domain.com/en, tenant1.domain.com/fr etc...

**Reply by thabaum** [reply]
This is all great stuff I will be sure to see if I can find this present in documentation and suggest to clarify things or add them as needed.

> Tenants should be able to change the look and feel of the page (change colors, set darktheme or even change/choose totally different theme or even customize one them selfs)
> "themes define the look and feel. The host user defines the themes which are available to sites."

I believe this would be possible through an extension of the Oqtane Framework handled by a theme and should be possible through profile settings selecting the styles to use depending.  I have given this area some thought and this would be the approach I believe I am going to take first to accomplish this.  I may need a module as well as the different theme styles to switch... not sure yet ;)

**Reply by sbwalker** [reply]
@thabaum it is already possible to create a theme which is customizable - this is the purpose of the Theme and Container Settings - it provides a UI so that users can customize the look/feel of a theme. The default Oqtane Theme already contains an example of how colors can be changed.

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/7c94793e-5eeb-41f7-82f3-ad75eb4af7b3)

**Reply by thabaum** [reply]
@sbwalker thanks for the tip, I will have to explore this feature further to see if it works with what I have in mind.
